source/opt/basic_block.h

```diff
@@ -118,6 +120,22 @@ class BasicBlock {
     return count > 0;
   }
 
+  // Return true if this block is a loop header block.
+  bool IsLoopHeader() const {
+    auto iItr = cend();
```

Comment:
could we just use "GetLoopMergeInst() != nullptr"?  The other issue is that all of this only works for structured control flow.  Will we want it to be more general than that?  How should we let OpenCL people know they cannot use this function?

Reply:
Done.

I've also asserted that we're processing a shader module in the SSA re-writer and the structured control compute.